### ISSUE TITLE
sh is not in $PATH

### ISSUE THREAD

**Issue by nascimento** [hadolint/hadolint#208]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior
Container was running fine with *sh*.

### Actual behavior

Receiving error when try to run sh.
Actualy I run the hadolint container inside GitlabCi and the default shell bin is SH.

Error:
docker: Error response from daemon: OCI runtime create failed: container_linux.go:348: starting container process caused "exec: \"sh\": executable file not found in $PATH": unknown.

### Steps to reproduce the behavior

```bash
docker run --rm hadolint/hadolint sh
```

**Comment by zemanlx** [comment]
We are in process of replacing base (`debian -> scratch`) for hadolint which is now in `master` branch = docker `latest` tag.

Please could you explain a bit more why you need shell?

**Comment by mrferos** [comment]
Wanted to chime in, currently we're using gitlab ci which by default will assume `sh` in path and will pipe the supplied commands to it:
```yaml
run-dockerfile-lint:
  image: hadolint/hadolint
  stage: test
  script:
    - hadolint - < Dockerfile
```

**Comment by lorenzo** [comment]
@mrferos is that behaviour configurable? It does not make a lot of sense to run a docker image by first running `sh` into it.

**Comment by mrferos** [comment]
@lorenzo reading the gitlab CI docs, I don't believe so. In the case of kubernetes executor (which is what we use) it spins up the desired image and then executes some default scripts ( https://docs.gitlab.com/runner/shells/#overview ) to clone the repo, set up default CI variables, etc then starts running the specified commands.

**Comment by lorenzo** [comment]
That's a very odd thing to assume. I would prefer not having to ship hadolint with a shell if possible. Let me see if there are any pointers around about this problem

**Comment by lorenzo** [comment]
@mrferos Ah ok, I understand why they make that assumption. You are registering the hadolint image as a runner, it is not just a command to execute but an actual container where all the tests will run.

My suggestion for now would be to make another image yourself that wraps hadolint and also contains a shell. We could add an additional image ourselves for a future version to support this use case and document the integration for gitlab.

**Comment by mrferos** [comment]
@lorenzo I'd be more than happy to submit a PR to build a debian-slim variant that builds off the base image that I see being done in `docker/hooks/build`

**Comment by lorenzo** [comment]
@mrferos please do!

**Comment by lorenzo** [comment]
@mrferos if you can add some notes about the new image in the `INTEGRATIONS` doc for gitlab I would also appreciate it

**Comment by mrferos** [comment]
PR is available [here](https://github.com/hadolint/hadolint/pull/210), though had I RTFM I would've seen the recommended way to run this in gitlab ci 🙃.

**Comment by lorenzo** [comment]
I'll close this as there is a documented solution for gitlab, and there is a pull request open for creating a separate docker image

**Comment by lwille** [comment]
just in case someone wants a quick solution: Have a look at [`erasys/hadolint`](https://hub.docker.com/r/erasys/hadolint/). 
Looking forward to see official Alpine/Debian builds!

**Comment by lorenzo** [comment]
@lwille did you see our recommendation for gitlab integration?

**Comment by lorenzo** [comment]
@lwille thanks, just wanted to know if you knew of that alternative already

**Comment by lorenzo** [comment]
The `hadolint/hadolint:latest-debian` image is now available for anyone who needs an official image with a shell

**Comment by nascimento** [comment]
Thank you guys
